src/consumption/elevate.js

```diff
@@ -0,0 +1,1827 @@
+ * @param {Object} certificateData - Certificate data for upload
+ * @param {String} solutionId - solutionId of the created solution
+ * @param {String} programId - programId of the created program
+ */
+async function insertCertificateTemplate(certificateData, solutionId, programId, loggedInUserId, userToken) {
+	const svgTemplateCreation = await createSvg(certificateData, loggedInUserId, userToken)
+	const baseTemplate = await checkCertificateBaseTemplate(certificateData)
+	const certificateDocument = {
+		status: common.STATUS_ACTIVE.toLowerCase(),
+		deleted: false,
+		solutionId,
+		programId,
+		baseTemplateId: baseTemplate._id,
+		createdAt: new Date(),
+		updatedAt: new Date(),
+		templateUrl: svgTemplateCreation.filePath,
+		issuer: { name: certificateData.issuer },
+		criteria: certificateData.criteria,
+	}
+
+	// Insert the template into the database
+	const certificateTemplateCollection = projectsMongoConnection.collection(
+		COLLECTIONS_MAP.get('CERTIFICATE_TEMPLATE')
+	)
+	const result = await certificateTemplateCollection.insertOne(certificateDocument)
+
+	// Validate the result of the template creation
+	if (!result || !result.insertedId) {
+		throw new Error(`Failed to insert the template into the ${COLLECTIONS.CERTIFICATE_TEMPLATE} collection.`)
+	}
+	// update the solution with the certificate template id
+	const solutionTemplateCollection = projectsMongoConnection.collection(COLLECTIONS_MAP.get('COLLECTIONS'))
+	const resultUpdateSolution = await solutionTemplateCollection.updateOne(
+		{ _id: solutionId },
+		{
+			$set: {
+				certificateTemplateId: result.insertedId,
+			},
+		}
+	)
+
+	// Validate the result of the template creation
+	if (!resultUpdateSolution) {
+		throw new Error(`Failed to update the template into the ${COLLECTIONS.SOLUTIONS} collection.`)
+	}
+	// update the template into projectTemplate collection
+	const projectTemplateCollection = projectsMongoConnection.collection(COLLECTIONS_MAP.get('TEMPLATES'))
+	const resultUpdateProjecTemplate = await projectTemplateCollection.updateOne(
+		{ solutionId },
+		{
+			$set: {
+				certificateTemplateId: result.insertedId,
+			},
+		}
+	)
+
+	// Validate the result of the template creation
+	if (!resultUpdateProjecTemplate.matchedCount) {
+		throw new Error(`No document found with solutionId: ${solutionId} in the ${COLLECTIONS.TEMPLATES} collection.`)
+	}
+
+	if (resultUpdateProjecTemplate.modifiedCount === 0) {
+		throw new Error(
+			`Document with solutionId: ${solutionId} was found but not updated in the ${COLLECTIONS.TEMPLATES} collection.`
+		)
+	}
+
+	return true
+}
+
+/**
+ * Create a duplicate solution from the given resource details
+ * @name duplicateResources
+ * @param {Object} resourceDetails - Object of resource details
+ * @param {String} created_by - created by user id
+ * @returns {Array} Array of objects of duplicate templates
+ */
+const duplicateResources = async (resourceDetails, resourceCertificate, programData) => {
```

Comment:
@adithyadinesh0412  Can we keep utils file for consumption or rearrange the functioin order, Main function keep at the top